holoviews/core/util.py

```diff
@@ -1163,22 +1163,26 @@ def arglexsort(arrays):
 
 def get_dynamic_item(map_obj, dimensions, key):
     """
-    Looks up an item in a DynamicMap given a list of dimensions
-    and a corresponding key. The dimensions must be a subset
-    of the map_obj key dimensions.
-    """
-    dmaps = map_obj.traverse(lambda x: x, ['DynamicMap'])
-    dmap = dmaps[0] if dmaps else map_obj
-    if key == () and (not dimensions or not dmap.kdims):
-        map_obj.traverse(lambda x: x[()], ['DynamicMap'])
-        return key, map_obj.map(lambda x: x.last, ['DynamicMap'])
-    elif isinstance(key, tuple):
-        dims = {d.name: k for d, k in zip(dimensions, key)
-                if d in map_obj.kdims}
-        key = tuple(dims.get(d.name) for d in map_obj.kdims)
-        el = map_obj.select(['DynamicMap', 'HoloMap'], **dims)
+    Traverses the supplied object selecting the requested key on
+    all HoloMap and DynamicMap objects.
+    """
```

Comment:
It would be nice to have a doctest style example here (even if it isn't a working one). Just be sure that it doesn't get picked up as one and then fail!

I understand the problem with a working doctest comes down to circular imports.